### PR TITLE
Merge with Upstream: mquandalle/meteor-jade

### PR DESCRIPTION
**Pull request by AlbinoGeek**
This PR would bring xiphias/meteor-jade-coffee in sync with mquandalle/meteor-jade
## The following file has to be modified still:
- packages/jade/plugin/handler.js

### CODE REVIEW

**Comment by AlbinoGeek** [comment]
This would close #3 and #4 .

---

It does not fix #1 or #2 however.

**Comment by AlbinoGeek** [comment]
Copied both packages into our live applications and there were no build or boot errors, no package or dependency errors; however: I still had issues #1 and #2 .

File: README.md [hunk 1]
```diff
@@ -381,7 +391,22 @@ This is an experimental version so there can be many unknown bugs, but the bigge
 ```
 meteor publish
 meteor admin get-machine os.linux.x86_64 --minutes 5
-meteor publish-for-arch xiphy:jade-compiler@0.5.0
-meteor publish-for-arch xiphy:jade-coffee@0.1.1
+meteor publish-for-arch xiphy:jade-compiler@0.5.2
```
Comment (xiphias) [inline README.md]: publish-for-arch was not needed, could you just leave meteor publish?

**Comment by AlbinoGeek** [comment]
Removed publish-for-arch lines from section `Publishing` in `README.md`

File: packages/jade-coffee/package.js [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
-  version: "0.1.4",
+  version: "0.1.6",
   name: "xiphy:jade-coffee",
   git: "https://github.com/xiphias/meteor-jade-coffee.git",
   documentation: "../../README.md"
 });
 
 Package.registerBuildPlugin({
-  name: "compileJade",
+  name: "compileJadeBatch",
```
Comment (xiphias) [inline packages/jade-coffee/package.js]: This doesn't work probably as there's no compileJadeBatch function yet, as we're using the old handler.js. The build shouldn't pass for this.

Comment (AlbinoGeek) [inline packages/jade-coffee/package.js]: Interesting, as neither version xiphy:master nor albinogeek:master will run entirely properly on my applications (giving the same "coffee run before jade ready" behavior.)  However; no build errors on Windows 10 with Meteor 1.2.1 .

I copied the `packages` directory into my meteor application, then did `meteor add xiphy:jade-coffee` and watched it build from local sources; including building compileJadeBatch just fine.

File: packages/jade/plugin/handler.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
+const path = Npm.require('path');
```
Comment (xiphias) [inline packages/jade/plugin/handler.js]: We're not creating jade plugin, could you take out this file?

File: History.md [hunk 4]
```diff
@@ -1,3 +1,14 @@
+## v0.4.5
+
+* Support Meteor 1.2 “Batch Plugin” API
```
Comment (xiphias) [inline History.md:3]: You could write
Not yet ported: Support Meteor 1.2 “Batch Plugin” API

File: packages/jade/plugin/handler.js (hunk 3 above)
Comment (AlbinoGeek) [inline packages/jade/plugin/handler.js]: Ahh okay, I was not sure what this file was for; hence the mention in the PR.

File: packages/jade-coffee/package.js (hunk 2 above)
Comment (xiphias) [inline packages/jade-coffee/package.js]: Did you run the jade-coffee test suite?

**Comment by AlbinoGeek** [comment]
Moment, I see a file that was not merged due to the naming (that's why there was a jade/plugin/handler.js ...)

File: packages/jade-coffee/package.js (hunk 2 above)
Comment (xiphias) [inline packages/jade-coffee/package.js]: Aha, I see this is not a function name..anyways it doesn't compile in Batch yet, could you revert to the old name? (compileJade)

Comment (AlbinoGeek) [inline packages/jade-coffee/package.js]: Can do; noticed an issue though: handler.js wasn't merged, I'm trying to merge it now.

**Comment by xiphias** [comment]
It's OK. Please leave handler.js to another PR as it's more complex to do.

File: packages/jade-coffee/package.js (hunk 2 above)
Comment (xiphias) [inline packages/jade-coffee/package.js]: Please don't merge handler.js in this PR, as it's a bigger structural
change.

Comment (AlbinoGeek) [inline packages/jade-coffee/package.js]: Fair point.

**Comment by AlbinoGeek** [comment]
OK;  and running tests again...

**Comment by xiphias** [comment]
Thanks, if both test suites run well, I'll merge the PR. Thanks very much!

**Comment by AlbinoGeek** [comment]
Which method of running tests previously worked for this project?  New README has new method

**Comment by AlbinoGeek** [comment]
New test syntax:

```
> meteor test-packages <exact package name>

# as in:

> meteor test-packages xiphy:jade-coffee
```

The docs say it loads a local one over one found online.

**Comment by AlbinoGeek** [comment]
```
>meteor remove mquandalle:jade
Changes to your project's package version selections:
mquandalle:jade           removed from your project
mquandalle:jade-compiler  removed from your project
mquandalle:jade: removed dependency

>meteor add xiphy:jade-coffee
xiphy:jade-compiler: updating npm dependencies -- jade...
compileJade: updating npm dependencies -- coffee-script...

Changes to your project's package version selections:
xiphy:jade-coffee    added, version 0.1.6
xiphy:jade-compiler  added, version 0.5.2

xiphy:jade-coffee: Jade template language with inline coffeescript and Javascript support

>meteor
[[[[[ ~ ]]]]]

=> Started proxy.
=> Started MongoDB.
=> Started your app.

=> App running at: http://localhost:3000/
   Type Control-C twice to stop.
```

Out of order loading is also not happening in one of our applications now :+1: .

### And as for the tests

```
>meteor test-packages xiphy:jade-coffee
[[[[[ Tests ]]]]]

=> Started proxy.
=> Started MongoDB.
=> Started your app.

=> App running at: http://localhost:3000/
   Type Control-C twice to stop.

(11 out of 11 passed , 56ms)
```

```
>meteor test-packages xiphy:jade-compiler
[[[[[ Tests ]]]]]

=> Started proxy.
=> Started MongoDB.
=> Started your app.

=> App running at: http://localhost:3000/
   Type Control-C twice to stop.

(5 out of 5 passed , 16ms)
```

**Comment by xiphias** [comment]
Sounds great.
I have these in the meteor packages generally:
autoupdate
jquery
underscore
test-in-browser
local-test:xiphy:jade-coffee
local-test:xiphy:jade-compiler

And just run meteor and see the results in the browser.
But it looks like what you did works as well.

**Pull request merged**